locators/locator.go

```diff
@@ -22,6 +23,7 @@ type Node struct {
 	PubKey  []byte
 	PrivKey []byte
 	Addr    ma.Multiaddr
+	Mux     sync.Mutex
```

Comment:
@stepanvalyavskiy why mutex is in `locators` package? I think it'd be more logically to put it into logging `package`.